wpimath/src/main/java/edu/wpi/first/math/controller/BangBangController.java

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.math.controller;
+
+import edu.wpi.first.math.MathSharedStore;
+import edu.wpi.first.math.MathUsageId;
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.util.sendable.SendableRegistry;
+
+/**
+ * Implements a bang-bang controller, which outputs either 0 or 1 depending on whether
+ * the measurement is less than the setpoint.  This maximally-aggressive control approach
+ * works very well for velocity control of high-inertia mechanisms,
+ * and poorly on most other things.
+ *
+ * <p>Note that this is an *asymmetric* bang-bang controller - it will not exert any control effort
+ * in the reverse direction (e.g. it won't try to slow down an over-speeding shooter wheel).  This
+ * asymmetry is *extremely important.*  Bang-bang control is extremely simple, but also
+ * potentially hazardous.  Always ensure that your motor controllers are set to "coast" before
+ * attempting to control them with a bang-bang controller.
+ */
+public class BangBangController implements Sendable {
+    public enum Direction {
+        kForward,
+        kReverse
+    }
+
+    private static int instances;
+
+    private final Direction m_direction;
+
+    private double m_tolerance;
+
+    private double m_setpoint;
+    private double m_measurement;
+
+    /**
+     * Creates a new bang-bang controller.
+     *
+     * <p>Always ensure that your motor controllers are set to "coast" before
+     * attempting to control them with a bang-bang controller.
+     *
+     * @param tolerance The controller's error tolerance, within which the controller
+     *                  will output zero.
+     * @param direction The direction of the bang-bang controller.
+     */
+    public BangBangController(
+        double tolerance,
+        Direction direction
```

Comment:
We don't have direction reversal on PIDController. We instead expect users to properly invert their encoders and motor controllers so positive is forward.